internal/loader/configuration_setting_loader.go

```diff
@@ -230,17 +245,26 @@ func (csl *ConfigurationSettingLoader) CreateKeyValueSettings(ctx context.Contex
 				}
 				switch *setting.ContentType {
 				case FeatureFlagContentType:
-					continue // ignore feature flag at this moment, will support it in later version
-				case KeyVaultReferenceContentType:
+					continue // ignore feature flag while getting key value settings
+				case SecretReferenceContentType:
 					if setting.Value == nil {
 						return nil, fmt.Errorf("The value of Key Vault reference '%s' is null", *setting.Key)
 					}
-					if csl.Spec.Secret == nil {
+
+					var secretType corev1.SecretType = corev1.SecretTypeOpaque
+					var err error
+					if secretTypeTag, ok := setting.Tags[PreservedSecretTypeTag]; ok {
+						secretType, err = parseSecretType(secretTypeTag)
+						if err != nil {
+							return nil, err
+						}
+					} else if csl.Spec.Secret == nil {
```

Comment:
Does the `else if` statement mean `Spec.Secret` can be `nil` when secret type is tls? If so, it may miss credentials when create `SecretReferenceResolver`.

Reply:
Good catch, updated